src/pykeen/models/unimodal/cp.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+"""Implementation of CP model."""
+
+from typing import Any, ClassVar, Mapping, Optional, Tuple, cast
+
+import torch
+
+from ..nbase import ERModel
+from ...constants import DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE
+from ...nn.emb import EmbeddingSpecification
+from ...nn.modules import CPInteraction
+from ...typing import Hint, Initializer, Normalizer
+
+__all__ = [
+    "CP",
+]
+
+
+class CP(ERModel):
+    r"""An implementation of CP as described in [lacroix2018]_ based on [hitchcock1927]_.
+
+    ---
+    name: Canonical Tensor Decomposition
+    citation:
+        author: Lacroix
+        year: 2018
+        arxiv: 1806.07297
+        link: https://arxiv.org/abs/1806.07297
+        github: facebookresearch/kbc
+    """
+
+    #: The default strategy for optimizing the model's hyper-parameters
+    hpo_default: ClassVar[Mapping[str, Any]] = dict(
+        embedding_dim=DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE,
+    )
+
+    def __init__(
+        self,
+        embedding_dim: int = 64,
+        rank: int = 64,
```

Comment:
need default HPO strategy for this

Reply:
50346de (I needed quite a bit to find out about how to specify the hpo range)